sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1314,7 +1317,7 @@ pi_int32 MapMemObject::enqueueImp() {
   *MDstPtr = MemoryManager::map(
       MSrcAllocaCmd->getSYCLMemObj(), MSrcAllocaCmd->getMemAllocation(), MQueue,
       MMapMode, MSrcReq.MDims, MSrcReq.MMemoryRange, MSrcReq.MAccessRange,
-      MSrcReq.MOffset, MSrcReq.MElemSize, std::move(RawEvents), Event);
+      MSrcReq.MOffset, MSrcReq.MElemSize, std::move(RawEvents), Event, MEvent);
```

Comment:
In scheduler, the only command type that is visible to the user via an event is ExecCGCommand. Others are internal, so there's no need to handle those (like MapMemObject, UnMapMemObject, MemCpyCommand, MemCpyCommandHost).

All of the CG cases in the `ExecCGCommand::enqueueImpQueue` do need to be handled though.

Reply:
Addressed.